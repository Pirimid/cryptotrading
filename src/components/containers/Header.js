import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import * as actions from '../../actions/headerActions';

export class Header extends React.Component {

 changePair = (pair) => {
  return (() => {
    this.props.actions.updateCurrencyPair( pair );
  }).bind(this);
};

  render() {
	return (
	  <div className="nav" >
		  <div className="navItems navlogo-container"><span className="navlogo" >Crypto Trade</span></div>
		  <div className="navItems pair-selection-container p-r has-dropdown">
			  <div className="pair-selection">
				  <div className="pair nav-title">{this.props.currentPair.unit1}/{this.props.currentPair.unit2}</div>
				  <div className="selection nav-sm-title">Select product</div>
			  </div>
			  <i className="fa arrow fa-chevron-down" aria-hidden="true"></i>
			  <div className="nav-dropdown">
					<h5 className="nav-title">BITCOIN(BTC)</h5>
					<ul className="nav-dropdown-list">
                        {this.props.appData.availablePairs.pairCategory1.map((object, i) => (
    						<li key={i} onClick={this.changePair(object)} >
    						    <a >
    						        <span>{object.unit1} / {object.unit2}</span>
    						        <span>{object.symbolUnit2} {object.lastTradePrice}</span>
    						        <span className={ object.last24HourPrice > 0 ? "up" : "down"}>{object.last24HourPrice}%</span>
                                </a>
                            </li>
                        ))}
					</ul>
			  </div>
		  </div>
		  <div className="navItems">
			  <div className="pair-selection">
				  <div className="pair nav-title">{this.props.currentPair.lastTradePrice} {this.props.currentPair.unit1}</div>
				  <div className="selection nav-sm-title">Last trade price</div>
			  </div>
		  </div>
		  <div className="navItems">
			  <div className="pair-selection">
				  <div className="pair nav-title green">{this.props.currentPair.last24HourPrice} %</div>
				  <div className="selection nav-sm-title">24 hour price</div>
			  </div>
		  </div>
		  <div className="navItems">
			  <div className="pair-selection">
				  <div className="pair nav-title">{this.props.currentPair.last24HourVolume} {this.props.currentPair.unit2}</div>
				  <div className="selection nav-sm-title">24 hour volume</div>
			  </div>
		  </div>
	  </div>
	);
  }
}

Header.propTypes = {
  actions: PropTypes.object.isRequired,
  currentPair: PropTypes.object.isRequired
};

function mapStateToProps(state) {
  return {
	currentPair: state.currentPair,
	appData: state.appData
  };
}

function mapDispatchToProps(dispatch) {
  return {
	actions: bindActionCreators(actions, dispatch)
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Header);
