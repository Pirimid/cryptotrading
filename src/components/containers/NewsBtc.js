import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {Grid, Row, Col, Image } from 'react-bootstrap';
import '../../styles/NewsBox.scss';
import * as actions from '../../actions/orderBookActions';

export class OrderTable extends React.Component {

  render() {
		return (
			<div>
				<h2 className="card-title">News BTC</h2>
				<div className="clearfix"></div>
				<div className="btc-news">
					<div className="card-news">
						<Row className="no-gutters">
							<Col md={2} className="text-center">
								<Image src="https://i.pinimg.com/236x/30/50/80/3050803135277ac1e153cc6aaeef796a--tech-news-first-time.jpg" className="news-thumbnail" responsive />
							</Col>
							<Col md={10}>
								<p className="p-text">
									How To Save on Bitcoin's Soaring Fees.
								</p>
							</Col>
						</Row>
					</div>
					<div className="card-news">
						<Row className="no-gutters">
							<Col md={2} className="text-center">
								<Image src="http://wire.kapitall.com/wp-content/image-import/ThinkstockPhotos-522646895-200x150.jpg" className="news-thumbnail" responsive />
							</Col>
							<Col md={10}>
								<p className="p-text">
									Bitcoin broker Coinbase booked $1 billion in revenue last year â€” so the company has told hovering VCs to back off
								</p>
							</Col>
						</Row>
					</div>
					<div className="card-news">
						<Row className="no-gutters">
							<Col md={2} className="text-center">
								<Image src="https://images.jagran.com/naidunia/bitcoin_plunge_2018118_154733_18_01_2018_m.jpg" className="news-thumbnail" responsive />
							</Col>
							<Col md={10}>
								<p className="p-text">
									Japan bitcoin bourse group stages Europe expansions
								</p>
							</Col>
						</Row>
					</div>
					<div className="card-news">
						<Row className="no-gutters">
							<Col md={2} className="text-center">
								<Image src="http://img.timesnownews.com/story/1514462803-bitcoin_new.jpg?d=200x150" className="news-thumbnail" responsive />
							</Col>
							<Col md={10}>
								<p className="p-text">
									Bitcoin To Drop As Low As $1000 This Year, Wall Street CIO Predicts
								</p>
							</Col>
						</Row>
					</div>
				</div>
			</div>
		);
  }
}

OrderTable.propTypes = {
  actions: PropTypes.object.isRequired,
  currentPair: PropTypes.object.isRequired
};

function mapStateToProps(state) {
  return {
	currentPair: state.currentPair
  };
}

function mapDispatchToProps(dispatch) {
  return {
	actions: bindActionCreators(actions, dispatch)
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(OrderTable);
