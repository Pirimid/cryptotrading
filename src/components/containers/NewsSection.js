import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as actions from '../../actions/orderBookActions';

export class NewsSection extends React.Component {

	render() {
		return (
			<div>
				<div className="card-fixed-header">
					<h2 className="card-title ">News</h2>
				</div>
				<div className="btc-news">
					<div className="news-item card-news">
						<a target="_blank" href="https://www.coindesk.com/cftc-sues-crypto-scheme-big-coin-fraud/" >
							CFTC Sues Obscure Crypto Scheme 'My Big Coin' for Fraud
						</a>
					</div>
					<div className="news-item card-news">
						<a target="_blank" href="https://www.coindesk.com/makerdao-quest-stable-stablecoin-continues/" >
							MakerDAO and More: The Quest for a Stable Stablecoin Continues
						</a>
					</div>
					<div className="news-item card-news">
						<a target="_blank" href="https://www.coindesk.com/building-a-base-bitcoin-demand-at-10k-hints-at-move-higher/" >
							Building a Base? Bitcoin Demand at $10K Hints at Move Higher
						</a>
					</div>
					<div className="news-item card-news">
						<a target="_blank" href="https://www.coindesk.com/stripe-to-end-support-for-bitcoin-transactions/" >
							Payment Processor Stripe to End Support for Bitcoin
						</a>
					</div>
					<div className="news-item card-news">
						<a target="_blank" href="https://www.coindesk.com/cftc-sues-crypto-scheme-big-coin-fraud/" >
							CFTC Sues Obscure Crypto Scheme 'My Big Coin' for Fraud
						</a>
					</div>
					<div className="news-item card-news">
						<a target="_blank" href="https://www.coindesk.com/makerdao-quest-stable-stablecoin-continues/" >
							MakerDAO and More: The Quest for a Stable Stablecoin Continues
						</a>
					</div>
					<div className="news-item card-news">
						<a target="_blank" href="https://www.coindesk.com/building-a-base-bitcoin-demand-at-10k-hints-at-move-higher/" >
							Building a Base? Bitcoin Demand at $10K Hints at Move Higher
						</a>
					</div>
					<div className="news-item card-news">
						<a target="_blank" href="https://www.coindesk.com/stripe-to-end-support-for-bitcoin-transactions/" >
							Payment Processor Stripe to End Support for Bitcoin
						</a>
					</div>
				</div>
			</div>
		);
	}
}

function mapStateToProps(state) {
	return {
		currentPair: state.currentPair
	};
}

function mapDispatchToProps(dispatch) {
	return {
		actions: bindActionCreators(actions, dispatch)
	};
}

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(NewsSection);
