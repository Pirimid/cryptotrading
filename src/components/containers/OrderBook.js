import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {Grid, Row, Col } from 'react-bootstrap';
import * as actions from '../../actions/orderBookActions';
import * as orderFormActions from '../../actions/orderFormActions';

export class OrderBook extends React.Component {

  constructor(props) {
    super(props);

    function onInterval(){
     this.props.actions.updateOrderBook( this.props.orderBook[Math.floor(Math.random() * 4)]);
   }
   setInterval( onInterval.bind(this) , 500);
 }

 handleMarketSizeClick = (value) => {
  return (() => {
    let orderForm = {
      activeTab : 'market',
      isBuy : true,
      buyAmount : value.marketSize * value.price ,
      buySize : value.marketSize,
      exchange : value.exchange
    };
    this.props.orderFormActions.updateOrderForm( orderForm );
  }).bind(this);
};

handlePriceClick = (value) => {
  return (() => {
    let orderForm = {
      activeTab : 'limit',
      isBuy : false,
      limitPrice : value.price,
      buyAmount : 0,
      buySize : 0,
      exchange : value.exchange
    };
    this.props.orderFormActions.updateOrderForm( orderForm );
  }).bind(this);
};

handleSpreadMarketSizeClick = (value) => {
  return (() => {
    let orderForm = {
      activeTab : 'market',
      isBuy : false,
      buyAmount : value.marketSize * value.price,
      buySize : value.marketSize,
      exchange : value.exchange
    };
    this.props.orderFormActions.updateOrderForm( orderForm );
  }).bind(this);

  this.props.actions.updateOrderFormExchange( value );
};

handleSpreadPriceClick = (value) => {
  return (() => {
    let orderForm = {
      activeTab : 'limit',
      isBuy : true,
      limitPrice : value.price,
      buyAmount : 0,
      buySize : 0,
      exchange : value.exchange
    };
    this.props.orderFormActions.updateOrderForm( orderForm );
  }).bind(this);
};

render() {
  return (
    <div className="card card-fh">
    <div className="card-fixed-header">
    <h2 className="card-title" data-card="orderbook">Order Book</h2>
    <div className="card-panel-header col-w-3">
    <label>Market Size</label>
    <label>Price({ this.props.currentPair.unit2 })</label>
    <label>Exchange</label>
    </div>
    </div>
    <div className="card-panel-body col-w-3">

    {this.props.orderBook.active.map((object, i) => (
      <div key={i} className="card-panel-body-rows">
        <span className="buy-size" onClick={this.handleMarketSizeClick(object)} >{object.marketSize}</span>
        <span className="buy-price down" onClick={this.handlePriceClick(object)} >{object.price}</span>
        <span>{object.exchange}</span>
      </div>))}

    <div className="card-panel-body-rows lbl">
    <label>{ this.props.currentPair.unit2 } Spread</label>
    <label>0.47</label>
    </div>
    {this.props.orderBook.active.map((object, i) => (
      <div key={i} className="card-panel-body-rows">
        <span className="ask-size" onClick={this.handleSpreadMarketSizeClick(object)} >{object.marketSize}</span>
        <span className="ask-price up" onClick={this.handleSpreadPriceClick(object)} >{object.price}</span>
        <span>{object.exchange}</span>
      </div>))}
    </div>
    </div>
    );
}
}

function mapStateToProps(state) {
  return {
    currentPair: state.currentPair,
    orderBook: state.orderBook
  };
}

function mapDispatchToProps(dispatch) {
  return {
    actions: bindActionCreators(actions, dispatch),
    orderFormActions: bindActionCreators(orderFormActions, dispatch)
 };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
  )(OrderBook);
