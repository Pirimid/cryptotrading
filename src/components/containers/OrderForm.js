import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {Grid, Row, Col, DropdownButton, MenuItem } from 'react-bootstrap';
import * as actions from '../../actions/orderFormActions';

export class OrderForm extends React.Component {

    handleTabClick = (event) => {
        this.props.actions.updateOrderFormTab( event.target.getAttribute('data-tab') );
    };

    handleBuySellToggle = () => {
        this.props.actions.toggleBuySellOrderForm();
    };


    handleExchangeSelection = (value) => {
        this.props.actions.updateOrderFormExchange( value );
    };


  render() {
		return (
		  	<div className="card card-order-form">
				<h2 className="card-title" data-card="orderform" >Order Form</h2>
				<div className="card-body">
					<div className="tabs">
						<ul className="tab-header-list">
							<li onClick={this.handleTabClick} data-tab='market' className={(this.props.orderForm.activeTab == 'market' ? 'active' : '')} >Market</li>
							<li onClick={this.handleTabClick} data-tab='limit' className={(this.props.orderForm.activeTab == 'limit' ? 'active' : '')} >Limit</li>
							<li onClick={this.handleTabClick} data-tab='stop' className={(this.props.orderForm.activeTab == 'stop' ? 'active' : '')} >Stop</li>
						</ul>
						<div className="clearfix"></div><br />
						<div className="tab-content">
							<button onClick={this.handleBuySellToggle} className={(this.props.orderForm.isBuy ? 'primary' : '') + " btn hw" } >Buy</button>
							<button onClick={this.handleBuySellToggle} className={(!this.props.orderForm.isBuy ? 'secondary' : '') + " btn hw" } >Sell</button>
							<div className="clearfix"></div><br />
							<small className="lbl-sm">Amount</small>
							<div className="form-input-group">
								<input type="text" className="form-input" value={this.props.orderForm.buyAmount} placeholder="0.00" />
								<span>{ !this.props.orderForm.isBuy || this.props.orderForm.activeTab == 'limit' ? this.props.currentPair.unit1 : this.props.currentPair.unit2 }</span>
							</div>
							<div className="clearfix"></div><br />
							<div className={(this.props.orderForm.activeTab == 'limit' ? 'show' : 'hide')}>
								<small className="lbl-sm">Limit Price</small>
								<div className="form-input-group">
									<input type="text" value={ this.props.orderForm.limitPrice } className="form-input" placeholder="0.00" />
									<span>{ this.props.currentPair.unit2 }</span>
								</div>
								<div className="clearfix"></div><br />
							</div>
							<div className={(this.props.orderForm.activeTab == 'stop' ? 'show' : 'hide')}>
								<small className="lbl-sm">Stop Price</small>
								<div className="form-input-group">
									<input type="text" className="form-input" placeholder="0.00" />
									<span>{ this.props.currentPair.unit2 }</span>
								</div>
								<div className="clearfix"></div><br />
							</div>
							<div className="sep-line"></div>
							<div className="clearfix">
								<div className="pull-left">
								    <label className="lbl-sm">Total
								        <small>({ this.props.orderForm.isBuy ? this.props.currentPair.unit1 : this.props.currentPair.unit2 })</small>=
                                    </label>
                                </div>
								<div className="pull-right"><label className="lbl-sm">{this.props.orderForm.buySize}</label></div>
							</div>
							<div className="clearfix"></div><br />
							<div className="clearfix">
								<div className="pull-left"><label className="lbl-sm">Exchange :</label></div>
								<div className="pull-right">
                                    <DropdownButton pullRight title={this.props.orderForm.exchange} id="exchange-selection-dropdown">
                                        {this.props.appData.exchanges.map((object, i) => (
                                            <MenuItem onSelect={this.handleExchangeSelection} eventKey={object}>{object}</MenuItem>
                                        ))}
                                    </DropdownButton>
                                </div>
							</div>
							<div className="clearfix"></div><br />
							<button className={( this.props.orderForm.isBuy ? 'primary' : 'secondary' ) + " btn block" } >Place Buy Order</button>
						</div>
						<br />
					</div>
				</div>
			</div>
		);
  }
}

function mapStateToProps(state) {
  return {
	currentPair: state.currentPair,
	orderForm : state.orderForm,
	appData : state.appData
  };
}

function mapDispatchToProps(dispatch) {
  return {
	actions: bindActionCreators(actions, dispatch)
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(OrderForm);
