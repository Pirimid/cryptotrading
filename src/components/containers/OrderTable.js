import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {Grid, Row, Col } from 'react-bootstrap';
import * as actions from '../../actions/fuelSavingsActions';

export class OrderTable extends React.Component {

  render() {
		return (
			<div>
				<h2 className="card-title dark-bg">Order Book</h2>
				<div className="clearfix"></div>
				<table className="table l-g-table">
					<thead>
						<tr>
							<th>Size</th>
							<th>Filled(BTC)</th>
							<th>Price</th>
							<th>Time</th>
							<th>Status</th>
						</tr>
					</thead>
					<tbody>
						<tr>
							<td>1</td>
							<td>Mark</td>
							<td>Otto</td>
							<td>sd</td>
							<td>@mdo</td>
						</tr>
						<tr>
							<td>2</td>
							<td>Jacob</td>
							<td>Thornton</td>
							<td>sd</td>
							<td>@fat</td>
						</tr>
						<tr>
							<td>3</td>
							<td>Larry the Bird</td>
							<td>@twitter</td>
							<td>@twitter</td>
							<td>sd</td>
						</tr>
					</tbody>
				</table>
			</div>
		);
  }
}

OrderTable.propTypes = {
  actions: PropTypes.object.isRequired,
  currentPair: PropTypes.object.isRequired
};

function mapStateToProps(state) {
  return {
	currentPair: state.currentPair
  };
}

function mapDispatchToProps(dispatch) {
  return {
	actions: bindActionCreators(actions, dispatch)
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(OrderTable);
