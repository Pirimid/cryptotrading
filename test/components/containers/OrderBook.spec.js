import React from "react";
import { mount, shallow } from "enzyme";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import { create } from "react-test-renderer";
import {OrderBook} from "../../../src/components/containers/OrderBook";
import ConnectedOrderBook from "../../../src/components/containers/OrderBook";
import initialState from "../../../src/reducers/initialState";

describe("<OrderBook />", () => {
  const actions = {
    updateCurrencyPair: jest.fn()
  };
  const store = configureMockStore()(initialState);

  it("should match snapshot", () => {
    const component = create(
      <Provider store={store}>
        <ConnectedOrderBook />
      </Provider>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should call updateCurrencyPair with selected pair", () => {
    // const wrapper = shallow(
    //   <OrderBook
    //     actions={actions}
    //     currentPair={initialState.currentPair}
    //     appData={initialState.appData}
    //   />
    // );

    // wrapper.find('li').first().simulate("click");
    // expect(actions.updateCurrencyPair).toHaveBeenCalledWith(
    //   initialState.appData.availablePairs.pairCategory1[0]
    // );
    
  });

});