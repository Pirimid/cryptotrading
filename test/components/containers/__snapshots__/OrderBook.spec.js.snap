// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderBook /> should match snapshot 1`] = `
<div
  className="card card-fh"
>
  <div
    className="card-fixed-header"
  >
    <h2
      className="card-title"
      data-card="orderbook"
    >
      Order Book
    </h2>
    <div
      className="card-panel-header col-w-3"
    >
      <label>
        Market Size
      </label>
      <label>
        Price(
        USD
        )
      </label>
      <label>
        Exchange
      </label>
    </div>
  </div>
  <div
    className="card-panel-body col-w-3"
  >
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.01879
      </span>
      <span
        className="down"
        onClick={[Function]}
      >
        11235.123356
      </span>
      <span>
        GDAX
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.02279837
      </span>
      <span
        className="down"
        onClick={[Function]}
      >
        11230.113344
      </span>
      <span>
        GDAX
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.05579837
      </span>
      <span
        className="down"
        onClick={[Function]}
      >
        11230.113322
      </span>
      <span>
        Bitfinex
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.01879
      </span>
      <span
        className="down"
        onClick={[Function]}
      >
        11229.113311
      </span>
      <span>
        Binance
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.02279837
      </span>
      <span
        className="down"
        onClick={[Function]}
      >
        11222.123344
      </span>
      <span>
        Bittrex
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.05579837
      </span>
      <span
        className="down"
        onClick={[Function]}
      >
        11212.123322
      </span>
      <span>
        Bitfinex
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.01879
      </span>
      <span
        className="down"
        onClick={[Function]}
      >
        11212.123316
      </span>
      <span>
        Krakan
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.02279837
      </span>
      <span
        className="down"
        onClick={[Function]}
      >
        11211.123344
      </span>
      <span>
        CEX.IO
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.05579837
      </span>
      <span
        className="down"
        onClick={[Function]}
      >
        11210.123322
      </span>
      <span>
        GDAX
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.01279837
      </span>
      <span
        className="down"
        onClick={[Function]}
      >
        11207.123312
      </span>
      <span>
        Bitfinex
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.13879837
      </span>
      <span
        className="down"
        onClick={[Function]}
      >
        11204.1233
      </span>
      <span>
        Krakan
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.01279837
      </span>
      <span
        className="down"
        onClick={[Function]}
      >
        11203.123312
      </span>
      <span>
        Bittrex
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.13879837
      </span>
      <span
        className="down"
        onClick={[Function]}
      >
        11201.1233
      </span>
      <span>
        HitBTC
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.01279837
      </span>
      <span
        className="down"
        onClick={[Function]}
      >
        11200.123312
      </span>
      <span>
        GDAX
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.01879
      </span>
      <span
        className="down"
        onClick={[Function]}
      >
        11200.123006
      </span>
      <span>
        Binance
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.02279837
      </span>
      <span
        className="down"
        onClick={[Function]}
      >
        11200.12
      </span>
      <span>
        GDAX
      </span>
    </div>
    <div
      className="card-panel-body-rows lbl"
    >
      <label>
        USD
         Spread
      </label>
      <label>
        0.47
      </label>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.01879
      </span>
      <span
        className="up"
        onClick={[Function]}
      >
        11235.123356
      </span>
      <span>
        GDAX
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.02279837
      </span>
      <span
        className="up"
        onClick={[Function]}
      >
        11230.113344
      </span>
      <span>
        GDAX
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.05579837
      </span>
      <span
        className="up"
        onClick={[Function]}
      >
        11230.113322
      </span>
      <span>
        Bitfinex
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.01879
      </span>
      <span
        className="up"
        onClick={[Function]}
      >
        11229.113311
      </span>
      <span>
        Binance
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.02279837
      </span>
      <span
        className="up"
        onClick={[Function]}
      >
        11222.123344
      </span>
      <span>
        Bittrex
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.05579837
      </span>
      <span
        className="up"
        onClick={[Function]}
      >
        11212.123322
      </span>
      <span>
        Bitfinex
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.01879
      </span>
      <span
        className="up"
        onClick={[Function]}
      >
        11212.123316
      </span>
      <span>
        Krakan
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.02279837
      </span>
      <span
        className="up"
        onClick={[Function]}
      >
        11211.123344
      </span>
      <span>
        CEX.IO
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.05579837
      </span>
      <span
        className="up"
        onClick={[Function]}
      >
        11210.123322
      </span>
      <span>
        GDAX
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.01279837
      </span>
      <span
        className="up"
        onClick={[Function]}
      >
        11207.123312
      </span>
      <span>
        Bitfinex
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.13879837
      </span>
      <span
        className="up"
        onClick={[Function]}
      >
        11204.1233
      </span>
      <span>
        Krakan
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.01279837
      </span>
      <span
        className="up"
        onClick={[Function]}
      >
        11203.123312
      </span>
      <span>
        Bittrex
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.13879837
      </span>
      <span
        className="up"
        onClick={[Function]}
      >
        11201.1233
      </span>
      <span>
        HitBTC
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.01279837
      </span>
      <span
        className="up"
        onClick={[Function]}
      >
        11200.123312
      </span>
      <span>
        GDAX
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.01879
      </span>
      <span
        className="up"
        onClick={[Function]}
      >
        11200.123006
      </span>
      <span>
        Binance
      </span>
    </div>
    <div
      className="card-panel-body-rows"
    >
      <span
        onClick={[Function]}
      >
        0.02279837
      </span>
      <span
        className="up"
        onClick={[Function]}
      >
        11200.12
      </span>
      <span>
        GDAX
      </span>
    </div>
  </div>
</div>
`;
